apisix/plugins/jwt-auth.lua

```diff
@@ -119,29 +143,74 @@ function _M.check_schema(conf, schema_type)
     if schema_type == core.schema.TYPE_CONSUMER then
         ok, err = core.schema.check(consumer_schema, conf)
     else
-        ok, err = core.schema.check(schema, conf)
+        return core.schema.check(schema, conf)
     end
 
     if not ok then
         return false, err
     end
 
-    if schema_type == core.schema.TYPE_CONSUMER then
-        if conf.algorithm ~= "RS256" and not conf.secret then
-            conf.secret = ngx_encode_base64(resty_random.bytes(32, true))
+    -- in nginx init_worker_by_lua context API calls are disabled,
+    -- also that is a costly operation during system startup.
+    if ngx.get_phase() == "init_worker" then
```

Comment:
We can check the data in Vault, but in the Admin API during creating. Sending a bundle of requests in full sync (which is not only happened during system startup) is too expensive. For instance, we don't connect to aws-lambda in the check schema. Even the data is valid, people can still change it later as the data does not come from the major data source.

Reply:
> We can check the data in Vault, but in the Admin API during creating. Sending a bundle of requests in full sync (which is not only happened during system startup) is too expensive

Yes, that's the current behaviour. For handling HTTP.PUT the ngx phase is `content_by_lua`. The full sync is done in apisix startup that is `init_worker_by_lua` phase. So in that case, the plugin simply skips the validation with external sources.
Btw, are you trying to convey something else?
> Even the data is valid, people can still change it later as the data does not come from the major data source.

Totally agree with you : )

Reply:
> The full sync is done in apisix startup that is init_worker_by_lua phase.

Not exactly. The full sync can happen elsewhere like the data in etcd is compacted.
BTW, even the sync timer is triggered in the init_worker_by_lua, it runs in a separate "timer" phase. Currently, there is no way to distinguish a DP side check_schema out of the CP side.

Even we have successfully checked the data, its benefit is limited. Maybe we can just leave a note in the doc?

Reply:
Resolved. Thanks